client/src/components/squids/SquidList.jsx

```diff
@@ -1,25 +1,62 @@
 import React from "react"
 
+import { Link } from "react-router-dom"
+
 import { useSquidList } from "../hooks/useSquidList"
 import { SquidTile } from "./SquidTile"
 
 import "../../style/index.pcss"
 
-export const SquidList = () => {
-  const squidListQuery = useSquidList()
+export const SquidList = (props) => {
+  const {
+    location: { search },
+  } = props
+
+  const pageOffset = parseInt(new URLSearchParams(search).get("page"), 10) - 1 || 0
+
+  const pageSize = 10
+  const squidListQuery = useSquidList(pageOffset, pageSize)
   const squids = squidListQuery.data || []
+  const lastPage = Math.ceil(squids.total / pageSize)
+
+  let firstPageLinkNumber
+  let lastPageLinkNumber
+  const pageNumberButtons = []
+
+  if (pageOffset < 3) {
```

Comment:
it may be worthwhile to extract a `Paginator` component to isolate some of this functionality